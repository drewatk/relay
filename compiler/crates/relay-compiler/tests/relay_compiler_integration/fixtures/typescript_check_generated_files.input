//- foo.ts
/**
 * @RelayResolver User.foo: RelayResolverValue
 */

//- bar.ts
graphql`fragment barFragment on User {
   foo
}`

//- relay.config.json
{
   "language": "typescript",
   "schema": "./schema.graphql",
   "eagerEsModules": true,
   "featureFlags": {
      "typescript_check_generated_files": true
   }
}

//- schema.graphql
type Query { user: User }
type User { foo: String }
