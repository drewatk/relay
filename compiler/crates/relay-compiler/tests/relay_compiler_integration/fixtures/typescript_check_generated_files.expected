==================================== INPUT ====================================
//- foo.ts
/**
 * @RelayResolver User.foo: RelayResolverValue
 */

//- bar.ts
graphql`fragment barFragment on User {
   foo
}`

//- relay.config.json
{
   "language": "typescript",
   "schema": "./schema.graphql",
   "eagerEsModules": true,
   "featureFlags": {
      "typescript_check_generated_files": true
   }
}

//- schema.graphql
type Query { user: User }
type User { foo: String }
==================================== OUTPUT ===================================
//- __generated__/barFragment.graphql.ts
/**
 * <auto-generated> SignedSource<<d5205fa7d07ed17c0e5eac12b44a7d44>>
 * @lightSyntaxTransform
 * @nogrep
 */

/* tslint:disable */
/* eslint-disable */

import { Fragment, ReaderFragment } from 'relay-runtime';
import { FragmentRefs } from "relay-runtime";
export type barFragment$data = {
  readonly foo: string | null | undefined;
  readonly " $fragmentType": "barFragment";
};
export type barFragment$key = {
  readonly " $data"?: barFragment$data;
  readonly " $fragmentSpreads": FragmentRefs<"barFragment">;
};

const node: ReaderFragment = {
  "argumentDefinitions": [],
  "kind": "Fragment",
  "metadata": null,
  "name": "barFragment",
  "selections": [
    {
      "alias": null,
      "args": null,
      "kind": "ScalarField",
      "name": "foo",
      "storageKey": null
    }
  ],
  "type": "User",
  "abstractKey": null
};

(node as any).hash = "f60f2dcc6b71a6c9ec170e68dc2c994d";

export default node;
